flow/hooks/track_operations.py

```diff
@@ -0,0 +1,164 @@
+# Copyright (c) 2023 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+"""Built in execution hook for basic tracking."""
+import json
+
+from .util import collect_metadata
+
+try:
+    import git
+except ImportError:
+    GIT = False
+else:
+    from .git_util import collect_metadata_with_git
+
+    GIT = True
+
+
+FN_LOGFILE = ".operations_log.txt"
+
+
+class TrackOperations:
+    """:class:`~.TrackOperations` tracks information about the execution of operations to a logfile.
+
+    This hooks can provides information on the start, successful completion, and/or
+    erroring of one or more operations in a `flow.FlowProject` instance. The logs are stored in a
+    file given by the parameter ``fn_logfile``. This file will be appended to if it already exists.
+    The default formating for the log provides the [time, job id, log level, and log message].
+    .. note::
+        All tracking is performed at the INFO level. To ensure outputs are captured in log files,
+        use the `--debug` flag when running or submitting jobs, or specify
+        `submit_options=--debug` in your directives (example shown below).
```

Comment:
Use two backticks for code font in Sphinx / rST. Also applies to some places below where code font should be used.
```suggestion
        use the ``--debug`` flag when running or submitting jobs, or specify
        ``submit_options=--debug`` in your directives (example shown below).
```